src/app/cli/CommandLineInterface.java

```diff
@@ -520,6 +541,295 @@ public void launch(){
+                        System.out.println("descend         makes sort into a descend if val=\"D\" otherwise stays ascended\n");
+                        continue;
+                    }
+                    ArrayList<Game> games_list= new ArrayList<>();
+                    if(tokens[2].equals("title")){
+                        Game[] game_array = app.search_game_name(tokens[1]);
+                        games_list = new ArrayList<>(Arrays.stream(game_array).toList());
+                    }
+                    else if (tokens[2].equals("platform")){
+                        Game[] game_array = app.search_game_platform(tokens[1]);
+                        games_list = new ArrayList<>(Arrays.stream(game_array).toList());
+                    }
+                    else if (tokens[2].equals("date")){
+                        Date release_date = new Date(Long.parseLong(tokens[1]));
+                        Game[] game_array = app.search_game_release_date(release_date);
+                        games_list = new ArrayList<>(Arrays.stream(game_array).toList());
+                    }
+                    else if (tokens[2].equals("developer")){
+                        Game[] game_array = app.search_game_developer(tokens[1]);
+                        games_list = new ArrayList<>(Arrays.stream(game_array).toList());
+                    }
+                    else if (tokens[2].equals("price")){
+                        Game[] game_array = app.search_game_price(tokens[1]);
+                        games_list = new ArrayList<>(Arrays.stream(game_array).toList());
+
+                    }
+                    else if (tokens[2].equals("genre")){
+                        Game[] game_array = app.search_game_genre(tokens[1]);
+                        games_list = new ArrayList<>(Arrays.stream(game_array).toList());
+                    }
+                    else {
+                        System.out.println("\nUsage: ");
+                        System.out.println("SEARCH_GAME { search_val | search_type | sort_val | descend }");
+                        System.out.println("search_val      value being searched");
+                        System.out.println("search_type     type that the being search is, below are acceptable inputs");
+                        System.out.println("                {title},{platform},{date},{developer},{price},{genre}");
+                        System.out.println("sort_val        optional argument to sort the results by, below are acceptable inputs");
+                        System.out.println("                {title},{price},{genre},{developer},{date}");
+                        System.out.println("descend         makes sort into a descend if val=\"D\" otherwise stays ascended\n");
+                        continue;
+                    }
+                    Comparator<Game> default_comparator = new Comparator<Game>() {
```

Comment:
comparators should be moved to `Game.java`, i'll do this myself at some point

Reply:
Ok, I still plan to fix the comparators in the CLI to allow the reversing to be implemented as stated in a different Comment.